Add lava tides cheat

TheChronographer opened the pull request:
Adds the 'lavalevel' cheat mirroring 'waterlevel'
### Work done

Format: `/lavalevel [level] [speed] [remainTime]` e.g. `/luarules lavalevel 200 7 120`

With no arguments progresses the current lava to the next tide Rhythm. If speed and remainTime are not specified: defaults to 0.25 elmo/frame and 1 second respectively.

#### Test 1
- Start any match with lava enabled, either the modoption waterislava or a map with lava.
- Enable `/cheats`
- Type `/luarules lavalevel 200 1 12`
Lava will skip the current tide rhythm and advance to a height of 200 for 12 seconds, then advance to the next tide rhythm. 

#### Test 2
- Start any match with lava enabled on a map with varying tide rhythm e.g. special hotstepper
- Enable `/cheats`
- Type `/luarules lavalevel`
Lava will skip the current tide rhythm and advance to the next tide rhythm in the standard sequence. 

#### Test 3
- Start any match with lava enabled.
- Enable `/cheats`
- Type `/luarules lavalevel 200`
Lava will skip the current tide rhythm and advance to a height of 200, slowly using the default speed of 0.25 and remain for 1 second, then progress the next tide rhythm. 

#### Test 4
- Start any match with lava enabled.
- Enable `/cheats`
- Type `/luarules lavalevel foo bar boo`
Lava not progress as this is invalid format.
 

### Screenshots:

<img width="1350" height="785" alt="image" src="https://github.com/user-attachments/assets/99bbe274-afc0-44ba-9da7-444d7fae8548" />

<img width="1634" height="1245" alt="image" src="https://github.com/user-attachments/assets/1593f0b9-59b4-4d9d-bd13-5b0685d82057" />

### To do:
- Are there other methods to call cheats or alias' to add. e.g. /luarules waterlevel -> /waterlevel

TheChronographer committed:
Add lavalevel cheat to insert a lava tide rhythm during a match.

TheChronographer committed:
Remove echos

The basics work now

TheChronographer committed:
Update map_lava.lua

TheChronographer committed:
Can i just get rid of these commented out parts?

TheChronographer committed:
Initial Commit

Adds the modoption and updates the generator for advanced settings input as string.

TheChronographer committed:
Check values

TheChronographer committed:
Input sanitising

TheChronographer committed:
Update lava.lua

TheChronographer committed:
New default to preserve map lava settings

Also remove superfluous validation check.

TheChronographer committed:
Update modoptions.lua

sprunk commented on luarules/gadgets/map_lava.lua:
```diff
@@ -260,6 +270,44 @@ if gadgetHandler:IsSyncedCode() then
+			local insertRemain = tonumber(tideParams[3])
+			if (insertLevel and insertlevel >= 0 and insertlevel % 1 ~= 0) and
+				(insertSpeed and insertspeed > 0) and 
+				(insertRemain and insertRemain > 0 and insertRemain *30 % 1 ~= 0) then
```
what's 30?

TheChronographer replied:
To land on a valid game frame. But yes you're right it should be defined as frame rate at the start.

TheChronographer commented on luarules/gadgets/map_lava.lua:
```diff
@@ -410,6 +460,23 @@ else  -- UNSYCNED
+	local function lavalevel(cmd, line, words, playerID)
+		-- lavalevel: Handles the '/lavalevel' chat command to adjust the lava level in-game.
+		-- Usage: /lavalevel [level] [speed] [remainTime], with no arguments progresses the current lava to the next tide Rhythm 
+		-- If speed and remainTime are not specified: defaults to 0.25 elmo/frame and 1 second respectively.
```
Consistency with usage in other areas. I.e. Map maker definitions definition the lava settings. 

Per second could make it more user friendly though. 0.1-0.3 -> 3-9?

sprunk replied:
Would be good to make all other places consistently use seconds too.

TheChronographer committed:
Update map_lava.lua

TheChronographer committed:
Removed magic 30s

Replaced game speed related constants with a single variable at the start

TheChronographer committed:
Change tide speed to elmo/sec

From elmo/frame.

The only map with an old map config is special hotstepper so I turned override on in the game config. This will need a map update at some point in the future and then it can be returned to override = false

TheChronographer committed:
rename to tweaklava

TheChronographer committed:
Merge branch 'Lava-Tides-Advanced-modoption' into Lava-tides-cheat

TheChronographer committed:
Merge branch 'master' into Lava-tides-cheat

TheChronographer committed:
Merge remote-tracking branch 'upstream/master' into Lava-tides-cheat

TheChronographer committed:
Clean up cheated tides from the loop

Added tooltip and remove the cheated tides after a loop

TheChronographer committed:
Update map_lava.lua
